dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,969 @@
+        }
+        List<Field> fields = asset.getContentType()
+            .fields();
+        for (Field field : fields) {
+            new FieldLoader().invalidate(field);
+        }
+    }
+
+
+
+    @Override
+    public InputStream writeObject(String id1, String id2, boolean live, String language, String filePath) {
+        try {
+
+            Identifier identifier = APILocator.getIdentifierAPI()
+                .find(id1);
+            Contentlet contentlet = null;
+            if (CacheLocator.getVeloctyResourceCache()
+                .isMiss(filePath)) {
+                if (LanguageWebAPI.canDefaultContentToDefaultLanguage()) {
+                    LanguageAPI langAPI = APILocator.getLanguageAPI();
+                    language = Long.toString(langAPI.getDefaultLanguage()
+                        .getId());
+                } else {
+                    throw new ResourceNotFoundException("Contentlet is a miss in the cache");
+                }
+            }
+
+            try {
+                contentlet = APILocator.getContentletAPI()
+                    .findContentletByIdentifier(identifier.getId(), live, new Long(language), APILocator.getUserAPI()
+                        .getSystemUser(), true);
+            } catch (DotContentletStateException e) {
+                contentlet = null;
+            }
+
+            if (contentlet == null || !InodeUtils.isSet(contentlet.getInode()) || contentlet.isArchived()) {
+
+                LanguageAPI langAPI = APILocator.getLanguageAPI();
+                long lid = langAPI.getDefaultLanguage()
+                    .getId();
+                if (lid != Long.parseLong(language)) {
+                    Contentlet cc = APILocator.getContentletAPI()
+                        .findContentletByIdentifier(identifier.getId(), live, lid, APILocator.getUserAPI()
+                            .getSystemUser(), true);
+                    if (cc != null && UtilMethods.isSet(cc.getInode()) && !cc.isArchived()
+                            && LanguageWebAPI.canApplyToAllLanguages(cc)) {
+                        contentlet = cc;
+                    } else {
+                        CacheLocator.getVeloctyResourceCache()
+                            .addMiss(filePath);
+                        throw new ResourceNotFoundException("Contentlet Velocity file not found:" + filePath);
+                    }
+                }
+            }
+
+            Logger.debug(this, "DotResourceLoader:\tWriting out contentlet inode = " + contentlet.getInode());
+            return buildVelocity(contentlet, identifier, !live, filePath);
+
+        } catch (Exception e) {
+            throw new ResourceNotFoundException("Contentlet Velocity file not found:" + filePath);
+        }
+    }
+
+
+
+    @Override
+    public void invalidate(Object obj) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void invalidate(Object obj, boolean live) {
+        // TODO Auto-generated method stub
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)